challenge_take_out_the_garbage/src/challenge_take_out_the_garbage/take_out.py

```diff
@@ -0,0 +1,191 @@
+        constructed (grabs the closest object in the volume of the surface)
+        """
+        smach.StateMachine.__init__(self, outcomes=["succeeded", "failed"])
+
+        # Create designators
+        self.empty_arm_designator = ds.UnoccupiedArmDesignator(robot.arms, robot.leftArm, name="empty_arm_designator")
+        self.grab_designator = ds.LockToId(robot=robot, to_be_locked=grab_designator)
+
+        with self:
+            @smach.cb_interface(outcomes=["locked"])
+            def lock(userdata=None):
+                """ 'Locks' a locking designator """
+                # This determines that self.current_item cannot not resolve to a new value until it is unlocked again.
+                self.grab_designator.lock()
+                if self.grab_designator.resolve():
+                    rospy.loginfo("Current_item is now locked to {0}".format(self.grab_designator.resolve().id))
+
+                return "locked"
+
+            smach.StateMachine.add("LOCK_ITEM",
+                                   smach.CBState(lock),
+                                   transitions={'locked': 'GRAB_ITEM'})
+
+            smach.StateMachine.add("GRAB_ITEM",
+                                   states.Grab(robot, self.grab_designator, self.empty_arm_designator),
+                                   transitions={'done': 'UNLOCK_ITEM_SUCCEED',
+                                                'failed': 'UNLOCK_ITEM_FAIL'})
+
+            @smach.cb_interface(outcomes=["unlocked"])
+            def unlock(userdata=None):
+                """ 'Unlocks' a locking designator """
+                self.grab_designator.unlock()
+
+                return "unlocked"
+
+            smach.StateMachine.add("UNLOCK_ITEM_SUCCEED",
+                                   smach.CBState(unlock),
+                                   transitions={'unlocked': 'succeeded'})
+
+            smach.StateMachine.add("UNLOCK_ITEM_FAIL",
+                                   smach.CBState(unlock),
+                                   transitions={'unlocked': 'failed'})
+
+
+class PlaceSingleItem(smach.State):
+    """ Tries to place an object. A 'place' statemachine is constructed dynamically since this makes it easier to
+     build a statemachine (have we succeeded in grasping the objects?)"""
+    def __init__(self, robot, place_designator):
+        """ Constructor
+
+        :param robot: robot object
+        :param place_designator: Designator that resolves to the pose to place at. E.g. an EmptySpotDesignator
+        """
+        smach.State.__init__(self, outcomes=["succeeded", "failed"])
+
+        self._robot = robot
+        self._place_designator = place_designator
+
+    def execute(self, userdata=None):
+
+        # ToDo: turn this into a proper statemachine
+        arm = None
+        # See if there's an arm holding something
+        for k, v in self._robot.arms.iteritems():
```

Comment:
I think this should be different nowadays, check with @alberth

Reply:
Indeed, robot has a `get_arm` where you can ask for an arm holding anything.